### PR TITLE
Assert HTTP responses and prettify errors.

### PR DESCRIPTION
**Pull request by jeschkies**
## High-level description

The current retrying/wait implementation waits for a condition to become
true. This is hard to debug in error cases. Even with extra logging one
has to find the correct log lines. This change introduces the use of
a different assertions for responses from HTTP requests. The error
prints the actual status code and body, e.g.

This should ease debugging since the JUnit XML file generated by pytest
will include this message at the top.

## Corresponding DC/OS tickets (obligatory)

  - [DCOS_OSS-2115](https://jira.mesosphere.com/browse/DCOS_OSS-2115)

### CODE REVIEW

**Comment by jeschkies** [comment]
@mesosphere-mergebot label Ready For Review

**Comment by orsenthil** [comment]
Hey @jeschkies - This is interesting! Look at the oddity `teamcity/dcos/test/azure/arm ` failed whereas the same test with onprem installer succeed.

I think, the stop_max_delay is coming to picture here and the failure is due to timeout. 

Since this test is succeeding in other platforms and if we could we re-run on Azure to get more stats on this.

**Comment by orsenthil** [comment]
@mesosphere-mergebot bump-ee

**Comment by orsenthil** [comment]
Hi @jeschkies  - Look at the status now, it is green! - https://teamcity.mesosphere.io/viewLog.html?buildId=971521&tab=buildResultsDiv&buildTypeId=DcOs_Open_Test_IntegrationTest_AzureArm

The failure was due to timeout (and yes, ofcourse a bug in marathon) to exhibit a flaky behavior at higher level.

**Comment by jeschkies** [comment]
@orsenthil, I've upgraded to the latest master. I think we saw a similar issue in our system integration tests as well. Good to know that we caught this earlier :).

**Comment by orsenthil** [comment]
@mesosphere-mergebot bump-ee

File: packages/dcos-integration-test/extra/test_networking.py [hunk 1]
```diff
@@ -62,13 +63,13 @@ def deploy(self, dcos_api_session):
...
-        retry_on_result=lambda res: res is False)
+        stop_max_delay=20 * 60 * 1000)
     def wait(self, dcos_api_session):
         r = dcos_api_session.marathon.get('v2/apps/{}'.format(self.id))
-        r.raise_for_status()
+        assert_response_ok(r)
+
         self._info = r.json()
-        return self._info['app']['tasksHealthy'] == self.app['instances']
+        assert self._info['app']['tasksHealthy'] == self.app['instances']
```
Comment (alenkacz) [inline packages/dcos-integration-test/extra/test_networking.py]: maybe a custom message here?

**Comment by jeschkies** [comment]
@mesosphere-mergebot ship_it_label_status_check

**Comment by jeschkies** [comment]
@mesosphere-mergebot label Ship It

**Comment by meichstedt** [comment]
@mesosphere-mergebot label Ship It

**Comment by orsenthil** [comment]
@jeschkies  - You notice `@mesosphere-teamcity
teamcity/dcos/test/aws/onprem/static` failure, That should be addressed first.

**Comment by orsenthil** [comment]
@mesosphere-mergebot override-status
teamcity/dcos/test/aws/onprem/static https://jira.mesosphere.com/browse/DCOS_OSS-2115

**Comment by orsenthil** [comment]
@mesosphere-mergebot bump-ee

**Comment by orsenthil** [comment]
@jeschkies  - can you bring this change into 1.11 (1.10) branch too? I was trying to cherry-pick. But instead, I think, this needs to be squashed can be brought in a single commit.

**Pull request merged**